### PR TITLE
Feature/unique wheres

### PR DESCRIPTION
**Pull request by pvsaintpe**
#3 
Extending creating unique indexes with conditions on `where`, example:

```php
Schema::create('cities', static function (Blueprint $table) {
       ... 
       $table->uniquePartial(['name', 'author_id'])
            ->where('editor_id', 1)
            ->whereRaw('field1 = ? and field2 <= ?', ['field1', 'field2'])
            ->whereNotNull('deleted_at')
            ->whereBetween('author_id', [1, 2, 3]);
      ...
```

PS. If you will use `$table->uniquePartial()` without extend methods after call `uniquePartial()`, it works as `$table->unique()`.

### CODE REVIEW

File: src/Compilers/UniqueWhereCompiler.php [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Database\Schema\Grammars\Grammar;
+use Illuminate\Support\Fluent;
+use InvalidArgumentException;
+
+class UniqueWhereCompiler
+{
+    public static function compile(Grammar $grammar, Blueprint $blueprint, Fluent $command): string
+    {
+        dd($blueprint, $command);
```
Comment (pvsaintpe) [inline src/Compilers/UniqueWhereCompiler.php]: It work in progress...

File: src/Schema/Definitions/UniqueWhereDefinition.php [hunk 2]
```diff
@@ -0,0 +1,124 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Umbrellio\Postgres\Schema\Definitions;
+
+use Illuminate\Support\Fluent;
+
+class UniqueWhereDefinition extends Fluent
```
Comment (lazeevv) [inline src/Schema/Definitions/UniqueWhereDefinition.php]: Im not sure, that this class should be in this namespace

Comment (pvsaintpe) [reply src/Schema/Definitions/UniqueWhereDefinition.php]: fixed

File: src/Schema/Definitions/UniqueWhereDefinition.php [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+{
+    /**
+     * @param string $sql
+     * @param array $bindings
+     * @param string $boolean
+     * @return UniqueWhereDefinition
+     */
+    public function whereRaw($sql, $bindings = [], $boolean = 'and')
+    {
+        return $this->compileWhere('raw', $boolean, compact('sql', 'bindings'));
```
[the remark this answers is not in the document]
Comment (pvsaintpe) [reply src/Schema/Definitions/UniqueWhereDefinition.php]: fixed

File: src/Schema/Definitions/UniqueWhereDefinition.php [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+        return $this->compileWhere('Basic', $boolean, compact('column', 'operator', 'value'));
+    }
+
+    /**
+     * @param $first
+     * @param mixed|string|null $operator
+     * @param mixed|string|null $second
+     * @param string $boolean
+     * @return UniqueWhereDefinition
+     */
```
[the remark this answers is not in the document]
Comment (pvsaintpe) [reply src/Schema/Definitions/UniqueWhereDefinition.php]: fixed

**Pull request merged**